pkg/lib/oauth/handler/service_token.go

```diff
@@ -58,6 +58,14 @@ func (s *TokenService) IssueOfflineGrant(
 		IDPSessionID: opts.IDPSessionID,
 		IdentityID:   opts.IdentityID,
 
+		// These fields exist in OfflineGrantRefreshToken too
+		// They need to be provided to handled cases that we refer to an offline grant without token hash
+		// Such as using an id_token
+		ClientID:        client.ClientID,
+		Scopes:          opts.Scopes,
+		AuthorizationID: opts.AuthorizationID,
+		TokenHash:       tokenHash,
+
```

Comment:
I suggest we do not write these fields anymore. There fields are read-only after this PR is merged.

I also have another suggestion. We rename these fields to private.

1. We can spot places still use these fields and correct them
2. Outside the package, we cannot initialize these fields. So they are read-only.

Reply:
However, there are cases that we need these fields. id token reference the offline grant without a token hash.

Reply:
We cannot make these fields private, because it can be deserialized from json. However, I will rename the field with prefix `ReadOnly__`